### PR TITLE
[New Rule] Suspicious Symbolic Link Created

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Identifies the creation of a symbolic link to a suspicious file or location. A symbolic link is a reference to a file or directory that acts as a pointer or shortcut, allowing users to access the target file or directory from a different location in the file system. An attacker can potentially leverage symbolic links for privilege escalation by tricking a privileged process into following the symbolic link to a sensitive file, giving the attacker access to data or capabilities they would not normally have.

## Detection
I am aware of the fact that [Potential Shadow File Read via Command Line Utilities](https://github.com/elastic/detection-rules/blob/main/rules/linux/privilege_escalation_shadow_file_read.toml) rule also captures symlinks to `/etc/shadow`, however, a symlink rule is just not complete without the addition of a symlink to `/etc/shadow`. In order to make the rule more complete, I chose to add it anyways. 

Redsector shows 0 hits in the last 365 days. AFAIK symlinks to /etc/shadow, shells, ssh keys, cron directories and other suspicious directories is not common behavior. Personally never did this in production, just during pentests or training. I Set the severity to low, and will monitor for potential FPs and tune accordingly.

```
process where host.os.type == "linux" and event.action : ("exec", "exec_event") and event.type == "start" and process.name == "ln" and 
(
/* suspicious files */
(process.args : ("/etc/shadow", "/etc/shadow-", "/etc/shadow~", "/etc/gshadow", "/etc/gshadow-") or 
(process.working_directory: "/etc" and process.args : ("shadow", "shadow-", "shadow~", "gshadow", "gshadow-"))) or 
/* suspicious bins */
(process.args : ("/bin/bash", "/bin/dash", "/bin/sh", "/bin/tcsh", "/bin/csh", "/bin/zsh", "/bin/ksh", "/bin/fish") or 
(process.working_directory : "/bin" and process.args : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) or 
(process.args : ("/usr/bin/bash", "/usr/bin/dash", "/usr/bin/sh", "/usr/bin/tcsh", "/usr/bin/csh", "/usr/bin/zsh", "/usr/bin/ksh", "/usr/bin/fish") or 
(process.working_directory : "/usr/bin" and process.args : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) or
/* suspicious locations */
(process.args : ("/etc/cron.d/*", "/etc/cron.daily/*", "/etc/cron.hourly/*", "/etc/cron.weekly/*", "/etc/cron.monthly/*")) or
(process.args : ("/home/*/.ssh/*", "/root/.ssh/*","/etc/sudoers.d/*", "/var/backups/*", "/dev/shm/*"))
) and not user.Ext.real.id : "0" and not group.Ext.real.id : "0"
```
image

### CODE REVIEW

File: rules/linux/privilege_escalation_linux_suspicious_symbolic_link.toml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+(process.working_directory: "/etc" and process.args : ("shadow", "shadow-", "shadow~", "gshadow", "gshadow-"))) or 
+/* suspicious bins */
+(process.args : ("/bin/bash", "/bin/dash", "/bin/sh", "/bin/tcsh", "/bin/csh", "/bin/zsh", "/bin/ksh", "/bin/fish") or 
+(process.working_directory : "/bin" and process.args : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) or 
+(process.args : ("/usr/bin/bash", "/usr/bin/dash", "/usr/bin/sh", "/usr/bin/tcsh", "/usr/bin/csh", "/usr/bin/zsh", "/usr/bin/ksh", "/usr/bin/fish") or 
+(process.working_directory : "/usr/bin" and process.args : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) or
+/* suspicious locations */
+(process.args : ("/etc/cron.d/*", "/etc/cron.daily/*", "/etc/cron.hourly/*", "/etc/cron.weekly/*", "/etc/cron.monthly/*")) or
+(process.args : ("/home/*/.ssh/*", "/root/.ssh/*","/etc/sudoers.d/*", "/var/backups/*", "/dev/shm/*"))
+) and not user.Ext.real.id : "0" and not group.Ext.real.id : "0"
```
Comment (brokensound77) [inline rules/linux/privilege_escalation_linux_suspicious_symbolic_link.toml:43]:
```suggestion
process where host.os.type == "linux" and event.action : ("exec", "exec_event") and event.type == "start" and process.name == "ln" and 
  (
    /* suspicious files */
    (process.args in ("/etc/shadow", "/etc/shadow-", "/etc/shadow~", "/etc/gshadow", "/etc/gshadow-") or 
    (process.working_directory == "/etc" and process.args in ("shadow", "shadow-", "shadow~", "gshadow", "gshadow-"))) or 
    /* suspicious bins */
    (process.args in ("/bin/bash", "/bin/dash", "/bin/sh", "/bin/tcsh", "/bin/csh", "/bin/zsh", "/bin/ksh", "/bin/fish") or 
    (process.working_directory in "/bin" and process.args : ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) 
or 
    (process.args in: ("/usr/bin/bash", "/usr/bin/dash", "/usr/bin/sh", "/usr/bin/tcsh", "/usr/bin/csh", "/usr/bin/zsh", "/usr/bin/ksh", "/usr/bin/fish") or 
    (process.working_directory == "/usr/bin" and process.args in ("bash", "dash", "sh", "tcsh", "csh", "zsh", "ksh", "fish"))) or
    /* suspicious locations */
    (process.args : ("/etc/cron.d/*", "/etc/cron.daily/*", "/etc/cron.hourly/*", "/etc/cron.weekly/*", "/etc/cron.monthly/*")) or
    (process.args : ("/home/*/.ssh/*", "/root/.ssh/*","/etc/sudoers.d/*", "/var/backups/*", "/dev/shm/*"))
  ) and 
  not user.Ext.real.id : "0" and not group.Ext.real.id : "0"
```

**Review by brokensound77 — COMMENTED** [review]
What are your thoughts on making this a building block rule?

**Comment by Aegrah** [comment]
@brokensound77 I was thinking of making this a BBR before I finished the query - but to me BBR's have their main use case in shipping DR's that might end up being very noisy, as more as a "last resort". This rule will (I am assuming) not be that noisy as all of the symlinks listed in this rule are *very* suspicious (also 0 hits in redsector last 365 days, and 0 in my clusters other than testing). In case this rule *would* turn out to generate a lot of noise, and those cannot be solved by some tunings, I would reconsider.

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**